### ISSUE TITLE
Linter throws exception when the only error is in the first line

### ISSUE THREAD

**Issue by lucasdf** [AtomLinter/linter-erlc#41]
There are two problems here. 

First, the linter throws an error if the only error or warning in the file is in the first line. To reproduce it, create a file named test.erl with the following content:
```
-module(not_test).
```
The following exception will occur:
```
[Linter] Error running erlc TypeError: Cannot read property 'lineNo' of null
    at _loop (init.js [sm]:110)
    at init.js [sm]:111
(anonymous) @ linter-registry.js [sm]:137
```

The second problem is that when the linter does not crash it will ignore any error or warning on the first line. For the same file named `test.erl`, the content below will display the undefined error on the second line but it will ignore the error on first line, which should display "Module name 'not_test' does not match file name".
```
-module(not_test).
-export([not_for/3]).
```

I will submit a PR to fix this soon.

**Comment by lucasdf** [comment]
The first problem (exception) was fixed by 552afb218aca3cb66222efa99536126af8a89e2e

**Comment by Arcanemagus** [comment]
Ah, cool. That's a side effect of moving to a functional style of iterating over that data since it's now checking whether the `match` is null instead of just assuming that it got a match.
